src/GAP/iso_gap_oscar.jl

```diff
@@ -0,0 +1,192 @@
+         return _iso_gap_oscar_field_finite_prime(F)
+       else
+         return _iso_gap_oscar_field_finite_ext(F)
+       end
+     else
+       if GAP.Globals.IsRationals(F)
+         return _iso_gap_oscar_field_rationals(F)
+       elseif GAP.Globals.IsCyclotomicCollection(F)
+         if GAP.Globals.IsCyclotomicField(F)
+           return _iso_gap_oscar_field_cyclotomic(F)
+         end
+       end
+     end
+   end
+
+   error("no method found")
+end
+
+# For finite prime fields, choose `GF(p)`
+# (of type `Nemo.GaloisField` if `p` is small (fits into `UInt64`),
+# `Nemo.GaloisFmpzField` otherwise)).
+function _iso_gap_oscar_field_finite_prime(F::GAP.GapObj)
+   p = characteristic(F)  # of type `fmpz`
+   if p < fmpz(2)^64
+     p = UInt64(p)
+   end
+   G = GF(p)
+   e = GAPWrap.One(F)
+
+   f(x) = G(fmpz(GAPWrap.IntFFE(x)))
+   finv(x::Union{Nemo.gfp_elem, Nemo.gfp_fmpz_elem}) = GAP.Obj(lift(x))*e
+
+   return MapFromFunc(f, finv, F, G)
+end
+
+# For finite non-prime fields, choose `FiniteField(pol, "z")`
+# (of type `FqNmodFiniteField` if `p` is small, `FqFiniteField` otherwise)
+# because we can choose the same polynomial as is used in GAP.
+function _iso_gap_oscar_field_finite_ext(F::GAP.GapObj)
+   p = characteristic(F)  # of type `fmpz`
+   if p < fmpz(2)^64
+     p = UInt64(p)
+   end
+   d = GAP.Globals.DegreeOverPrimeField(F)
+   e = GAPWrap.One(F)
+
+   R, x = PolynomialRing(GF(p), "x")
+   if GAP.Globals.IsPrimeField(GAP.Globals.LeftActingDomain(F))
+     pol_gap = GAP.Globals.DefiningPolynomial(F)
+     if GAPWrap.IsAlgebraicExtension(F)
+#FIXME:
+# As soon as the problem from https://github.com/gap-system/gap/issues/4694
+# is fixed, go back to `Basis_F = GAP.Globals.Basis(F)` also in this case.
+# Note that we want a basis in the filter `IsCanonicalBasisAlgebraicExtension`.
+        Basis_F = GAP.Globals.Objectify(GAP.Globals.NewType(
+                  GAP.Globals.FamilyObj(F),
+                  GAP.Globals.IsCanonicalBasisAlgebraicExtension),
+                  GAP.NewPrecord(0))
+        GAP.Globals.SetUnderlyingLeftModule(Basis_F, F)
+     else
+        Basis_F = GAP.Globals.Basis(F)
+     end
+   else
+     # The GAP field is not an extension of the prime field.
+     # What is a reasonable way to compute (on the GAP side) a polynomial
+     # w.r.t. the prime field, and to decompose field elements w.r.t.
+     # the corresponding basis?
+     error("extensions of extension fields are not supported")
+   end
+
+   coeffs = GAP.Globals.CoefficientsOfUnivariatePolynomial(pol_gap)
+   coeffs = [fmpz(Oscar.GAPWrap.IntFFE(x)) for x in coeffs]
+   pol = R(coeffs)
+   G = FiniteField(pol, "z")[1]
+   Basis_G = Vector{elem_type(G)}(undef, d)
+   Basis_G[1] = G(1)
+   for i = 2:d
+      Basis_G[i] = Basis_G[i - 1]*gen(G)
+   end
+
+   function f(x)
+      v = GAPWrap.Coefficients(Basis_F, x)
+      v_int = [fmpz(GAPWrap.IntFFE(v[i])) for i = 1:length(v)]
+      return sum([v_int[i]*Basis_G[i] for i = 1:d])
```

Comment:
We have `length(v) == d`, right? If so, perhaps the following avoids an allocation (or perhaps Julia is already clever enough to do it for us?)
```suggestion
      return sum([fmpz(GAPWrap.IntFFE(v[i])*Basis_G[i] for i = 1:d])
```

But isn't there a constructor for finite field elements which directly takes a coefficient list? That ought to be even more efficient (depending on what `basis(G)` returns... hopefully just what we want... but this doesn't seem to be documented anywhere, in fact `basis(::FinField)` seem undocumented??? @fieker @thofma?

Reply:
Correct. There are no constructors taking a list of coefficients and the basis function is also a crouch :)